app/webpacker/controllers/cancel_orders_controller.js

```diff
@@ -0,0 +1,30 @@
+import ApplicationController from "./application_controller";
+
+export default class extends ApplicationController {
+  connect() {
+    super.connect();
+  }
+
+  confirm() {
+    const send_cancellation_email = document.querySelector(
+      "#send_cancellation_email"
+    ).checked;
+    const restock_items = document.querySelector("#restock_items").checked;
+    const order_ids = [];
+
+    document
+      .querySelectorAll("#listing_orders input[name='order_ids[]']:checked")
+      .forEach((checkbox) => {
```

Comment:
Is this possible to use `target` (for `send_cancellation_email`, `restock_items `, ...)  instead of the whole document via `document.querySelector`?
Every time it's possible, I'd prefer this since we are not too strongly tied to DOM structure.

Reply:
I have not access to the checkboxes by using target.
Controller is called from within the modal and its scope does not contain checkboxes;

This is why I used a DOM query selector just like you have done in #9804

I understand the need, so I tried to use a new feature of StimulusLS: outlets.
I have played a bit with outlets: I have added an outlet attribute in confirm_modal_component.haml.html.
But it does not work and for now, name of this attribute must be the same as the controller.
From what I have read on the StimulusJS issues, this features is still in development. Furthermore, it would need to be wrapped in a  conditional. 
And least, you have to use an outlet with a CSS selector.

Another idea if you want to keep the modal would be instead of calling a stimulus controller from it, to instead trigger a custom event on the tbody of the lines, and a an attribute with each and every controller needed. (But we might calling many ctrls at once?)
But it is only an idea, I have not tested it and it would mean I could not reuse the code from #9804

Reply:
Ok, now I see. That's clear. 
Thanks for letting me know about [outlets](https://stimulus.hotwired.dev/reference/outlets), I didn't know about it. Maybe something I should use soon. 

Maybe for that particular case, it's ok to do some querying with `document.querySelectorAll()`. TBH I'm not convinced about how things (or we?) handle such a thing with StimulusJS. 

Conclusion: I think that's perfect for now! Thanks! 🙏

Reply:
Thank you @jibees

I think the stack(StimulusJS/StimulusReflex/ViewComponent/ViewComponentReflex) is new and wide.

You can access to the back in StimulusReflax and also in ViewComponentReflex, so which Strategy is the better, is a bit hard to know. Also when you have to handle both some modal + back-end code can be tricky because you want access both to front-end and back-end :)
I think the way you designed it at first is a good way to factor. The part I had been trying to factor is the bottom lines of the view where are the modals for each choice that cannot be factored for now.